lib/hammer_cli_foreman/compute_profile.rb

```diff
@@ -0,0 +1,412 @@
+            h.section name do |h|
+              if defined?(provider.compute_profile_create_help)
+                provider.compute_profile_create_help(h)
+              else
+                provider.host_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name )
+        validator.any(:option_compute_resource_id, :option_compute_resource_name)
+      end
+
+      def request_params
+        params = super
+        params['compute_attribute']['vm_attrs'] = option_compute_attributes || {}
+        if option_interface
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'] = {Time.now.to_i.to_s => option_interface}
+        end
+        if option_volume
+            params['compute_attribute']['vm_attrs']['volumes_attributes'] = {Time.now.to_i.to_s => option_volume}
+        end
+        params
+      end
+
+      success_message _('Compute profile attributes are set.')
+      failure_message _('Could not set the compute profile attributes')
+      build_options
+    end
+
+    class UpdateAttributes < HammerCLIForeman::UpdateCommand
+      command_name 'update-attributes'
+      resource :compute_attributes
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface-id', 'INTERFACE_ID', _('Interface id'),
+             :format => HammerCLI::Options::Normalizers::Number.new
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--volume-id', 'VOLUME_ID', _('Volume id'),
+             :format => HammerCLI::Options::Normalizers::Number.new
+
+      extend_help do |h|
+        h.section _('Provider specific options') do |h|
+          ::HammerCLIForeman::ComputeProfile.compute_resources.each do |name, provider|
+            h.section name do |h|
+              if defined?(provider.compute_profile_create_help)
+                provider.compute_profile_create_help(h)
+              else
+                provider.host_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name ).required
+        validator.any(:option_compute_resource_id, :option_compute_resource_name).required
+        validator.any(:option_interface, :option_volume, :option_compute_attributes).required
```

Comment:
How do I identify what compute attribute I'm going to update? I'd expect `--compute-attribute` or `--compute-attribute-id`. 
Is compute-attribute-name unique or do we need to add also compute profile or compute resource to search for the compute attribute by name? If so It should be required only when there is compute attribute name given.

Reply:
Or is it that (compute profile,  compute resource) has at most one compute attribute?

Reply:
compute attribute is a combination of compute profile and compute resource. It doesn't have a name